src/sdl2_compat.c

```diff
@@ -1333,6 +1334,31 @@ SDL_LOG_IMPL(Error, ERROR)
 SDL_LOG_IMPL(Critical, CRITICAL)
 #undef SDL_LOG_IMPL
 
+static void SDLCALL LogOutputFunction3to2(void *userdata, int category, SDL_LogPriority priority, const char *message)
+{
+    LogOutputFunction2(userdata, category, LogPriority3to2(priority), message);
+}
+
+SDL_DECLSPEC void SDLCALL SDL_LogSetOutputFunction(SDL2_LogOutputFunction callback, void *userdata)
+{
+    if ((SDL_LogOutputFunction)callback == SDL3_GetDefaultLogOutputFunction()) {
+        /* This is the original SDL3 default logger. Don't use the 3to2 thunk. */
+        SDL3_SetLogOutputFunction((SDL_LogOutputFunction)callback, userdata);
+    } else {
+        LogOutputFunction2 = callback;
+        SDL3_SetLogOutputFunction(LogOutputFunction3to2, userdata);
+    }
+}
+
+SDL_DECLSPEC void SDLCALL SDL_LogGetOutputFunction(SDL2_LogOutputFunction *callback, void **userdata)
+{
+    SDL3_GetLogOutputFunction((SDL_LogOutputFunction *)callback, userdata);
+    if (callback && (SDL_LogOutputFunction)*callback == LogOutputFunction3to2) {
```

Comment:
It's not explicitly documented that you can pass NULL for one of these parameters, but real SDL2 lets you so I handle that too.